src/dropdown.ts

```diff
@@ -2664,60 +2693,102 @@ export default class Dropdown extends LitElement implements FormControl {
+            // they often aren't. So we avoid throwing unless we have to. And removing the value would,
+            // in a small way, be a breach of Dropdown's contract with consumers. So we enable the option.
+            event.target.disabled = false;
+          }
+
+          if (isOptionNewlySelected) {
+            this.selectedAndEnabledOptions = [
+              ...this.selectedAndEnabledOptions,
+              event.target,
+            ];
+
+            if (!this.#isSelectionFromValueSetter) {
+              this.#value = [...this.value, event.target.value];
+            }
+          }
+        } else {
+          if (!this.#isSelectionFromValueSetter) {
+            this.#value = this.#value.filter((_, index) => {
+              return (
+                index !==
+                this.selectedAndEnabledOptions.indexOf(
+                  event.target as DropdownOption,
+                )
+              );
+            });
           }
+
+          this.selectedAndEnabledOptions =
+            this.selectedAndEnabledOptions.filter(
+              (option) => option !== event.target,
+            );
         }
+      } else if (!this.multiple) {
+        if (event.target.selected) {
+          if (event.target.disabled) {
+            event.target.disabled = false;
+          }
 
-        this.#value = event.target.value ? [event.target.value] : [];
+          if (isOptionNewlySelected) {
+            this.selectedAndEnabledOptions = [
+              ...this.selectedAndEnabledOptions,
+              event.target,
+            ];
+          }
+
+          for (const option of this.#optionElements) {
+            if (option !== event.target) {
+              option.selected = false;
+            }
+          }
+
+          if (!this.#isSelectionFromValueSetter) {
+            this.#value = event.target.value ? [event.target.value] : [];
+          }
+        } else {
+          this.selectedAndEnabledOptions =
+            this.selectedAndEnabledOptions.filter(
+              (option) => option !== event.target,
+            );
+
+          if (!this.#isSelectionFromValueSetter) {
+            this.#value = this.lastSelectedAndEnabledOption?.value
+              ? [this.lastSelectedAndEnabledOption.value]
+              : [];
+          }
+        }
       }
     }
 
-    // If single-select, Dropdown's internal label now needs to be updated to reflect the
-    // selected option(s). `this.internalLabel` uses the `this.lastSelectedAndEnabledOption` getter,
-    // whose return value is derived from the state of another component: Dropdown Option.
-    // That component's state is reactive. But a change to it won't trigger a rerender of
-    // this component.
-    //
-    // If multiselect, the `areSomeOptionsSelected` and `areAllOptionsSelected` getters need
-    // to rerun so Select All is updated. So do Dropdown's tags.
-    this.requestUpdate();
```

Comment:
No longer needed now that `this.selectedAndEnabledOptions`, which triggers a rerender, is set above.